### PR TITLE
Add tests for other webhook types

### PR DESCRIPTION
**Pull request by adammw**
Adding test-table-driven-tests for TestReconcileWebhook providing a resource of each type.

### CODE REVIEW

**Comment by maxsmythe** [comment]
Thanks for writing the tests! It looks like the gomofules need to be updated.

**Comment by adammw** [comment]
updated

**Comment by maxsmythe** [comment]
It looks like there are merge conflicts on the go.mod file?

File: pkg/rotator/rotator_test.go [hunk 1]
```diff
@@ -9,7 +9,10 @@ import (
 	"github.com/onsi/gomega"
 	admissionv1 "k8s.io/api/admissionregistration/v1"
 	corev1 "k8s.io/api/core/v1"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
```
[the remark this answers is not in the document]
Comment (adammw) [reply pkg/rotator/rotator_test.go]: Rebased onto #39

**Review by maxsmythe — APPROVED** [review]
LGTM

Should we close #39 now that this PR is rebased on it?

**Review by sozercan — APPROVED** [review]
lgtm

**Pull request merged**